README.md

```diff
@@ -2,81 +2,134 @@
 
 Auto completion plugin for nvim.
 
-
-# Concept
+## Table Of Contents
+- [Concept](#concept)
+- [Usage](#usage)
+  - [Prerequisite](#prerequisite)
+  - [Configuring in Vimscript](#configuring-in-vimscript)
+  - [Configuring in Lua](#configuring-in-lua)
+  - [Mappings](#mappings)
+  - [Source Configuration](#source-configuration)
+- [Builtin Sources](#builtin-sources)
+- [Development](#development)
+  - [Example Source](#example-source)
+  - [The Source](#the-source)
+  - [Public API](#public-api)
+    - [Vimscript](#vim-script)
+    - [Lua](#lua)
+
+
+## Concept
 
 - Lua source & Vim source
 - Better matching algorithm
 - Support LSP completion features (trigger character, isIncomplete)
 
 
-# Usage
+## Usage
 
 The `source` option is required but others can be omitted.
 
-```viml
+### Prerequisite
+You must set `completeopt` to `menu,menuone,noselect` which can be easily done as follows.
 
-" You must set this option.
+Using Vimscript
+```viml
 set completeopt=menu,menuone,noselect
+```
+or using Lua
+```lua
+vim.o.completeopt = "menu,menuone,noselect"
+```
+
+### Configuring in Vimscript
+
+If you're using `init.vim`, here are the available options for `nvim-compe`.
+
+```viml
+let g:compe = {}
+let g:compe.enabled = v:true " whether or not nvim-compe is enabled. default: `true`.
+let g:compe.debug = v:false " whether or not nvim-compe should display debug info. default: `false`
+let g:compe.min_length = 1 " minimal length to trigger completion. default: `1`
+" controls nvim-compe preselect behaviour. default: `enable`
+" enable: preselect completion item only if the source told nvim-compe to do so. Eg. completion from `gopls`
+" disable: never preselect completion item regardless of source
+" always: always preselect completion item regardless of source
+let g:compe.preselect = 'enable' " default: `enable`
+let g:compe.throttle_time = 80 " throttle nvim-compe completion menu. default: `80`
+let g:compe.source_timeout = 200 " timeout for nvim-compe to get completion items. default: `200`
+let g:compe.incomplete_delay = 400 " delay for LSP's `isIncomplete`. default: `400`
+let g:compe.allow_prefix_unmatch = v:false
+
+-- define your nvim-compe sources here
+-- you MUST fill this field
+let g:compe.source = {}
+let g:compe.source.path = v:true " path completion. default: `false`
+let g:compe.source.buffer = v:true " buffer completion. default: `false`
+let g:compe.source.vsnip = v:true " vsnip completion, make sure you have `vim-vsnip` installed. default: `false`
+let g:compe.source.nvim_lsp = v:true " nvim builtin LSP completion. default: `false`
+let g:compe.source.nvim_lua = v:true " nvim's lua stdlib completion. default: `false`
+let g:compe.source.your_awesome_source = {
+  " you can also override completion source configuration
+  " by setting it to a `table` instead of `boolean`
+  " see `Source Configuration` for more information
+  \}
```

Comment:
Sorry. I don't want to maintein same explanation with Lua example and VimL example both.
Can you remove this duplication?

Reply:
sure

Reply:
It would be good to add a separate section. Thank you.
I don't stick to comment for explanation.